[addons] move GetTempAddonBasePath() from binary to normal AddonManager

AlwinEsch opened the pull request:
## Description
This is the first step to remove the CBinaryAddonManager. Since removal
of cpluff this no more direct needed.

Therefore this request, alone, since this is currently used for the binaries, I also want to ask whether there is something similar in the ***Python*** addons, which gives them a temporary path?
If so, we could combine the two so that there is more order in the mess.

My goal is to make the addon interface cleaner and then try to make it more stable. 
Because there is so much mix between things at the moment, it becomes difficult to create a centrally controlled path that takes care of what (e.g. ongoing addon changes while Kodi is running...)

I may also change the name "GetTempAddonBasePath" to something else. Can you think of something or do you think it's OK?

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

garbear commented on xbmc/addons/AddonManager.cpp:
```diff
@@ -131,6 +131,10 @@ bool CAddonMgr::Init()
 {
   m_database.Close();
+
+  /* If temporary directory was used from addon delete them */
```
```c++
/* If temporary directory was used from add-on, delete it */
```

garbear commented on xbmc/addons/AddonManager.h:
```diff
@@ -238,6 +238,16 @@ namespace ADDON
     const AddonInfoPtr GetAddonInfo(const std::string& id, TYPE type = ADDON_UNKNOWN);
 
+    /*!
+     * @brief To get the path where temporary addon parts can be becomes stored
```
```c++
* @brief Get the path where temporary add-on files are stored
```

garbear commented on xbmc/addons/AddonManager.h:
```diff
@@ -238,6 +238,16 @@ namespace ADDON
+     *
+     * @return the base path used for temporary addon paths
+     *
+     * @warning the folder and his contents becomes deleted during stop and
```
```
* @warning the folder and its contents are deleted when Kodi is closed
```

garbear commented on xbmc/addons/AddonManager.h:
```diff
@@ -258,6 +268,10 @@ namespace ADDON
     std::set<std::string> m_optionalAddons;
     ADDON_INFO_LIST m_installedAddons;
+
+    // Temporary path where can given to addons, his content becomes during
```
```
// Temporary path given to add-ons, whose content is deleted when Kodi is stopped
```

AlwinEsch replied:
Thanks a lot, my bad english is corrected :smile: !

garbear reviewed (approved):
Left some grammar comments, and the rest looks good

AlwinEsch committed:
[addons] move GetTempAddonBasePath() from binary to normal AddonManager

This is the first step to remove the CBinaryAddonManager. Since removal
of cpluff this no more direct needed.

ronie commented:
> I also want to ask whether there is something similar in the _**Python**_ addons, which gives them a temporary path?

nope, there's no such thing as far as i know.

AlwinEsch commented:
Thanks to you @ronie, would it make sense to have something there too?

AlwinEsch commented:
e.g. to have:
```py
tmpPath = self.Addon.getTempAddonPath()
```
like this http://alwinesch.github.io/group__cpp__kodi.html#ga437ac949d8c4dbe95e9cb04830dd6ff8

ronie commented:
> Thanks to you @ronie, would it make sense to have something there too?

we always tell python addon devs to store all their (temp) data in ../userdata/addon_data/addon.id
afaik there's no need for anything else..

AlwinEsch commented:
jenkins build this please
